CMakeLists.txt

```diff
@@ -303,25 +306,26 @@ if (${ENABLE_DOC})
     )
 endif (${ENABLE_DOC})
 
-# compile executables
-file( GLOB binaries "${CMAKE_SOURCE_DIR}/examples/*.cc")
-foreach(binaryfile ${binaries})
-  get_filename_component(binaryname  ${binaryfile} NAME_WE)
-  add_executable(${binaryname} ${binaryfile})
-  target_link_libraries(${binaryname} "${LIBRASCAL_NAME}")
-endforeach(binaryfile ${binaries})
-configure_file("${CMAKE_SOURCE_DIR}/examples/crystal_structure.json" ${CMAKE_BINARY_DIR} COPYONLY)
-configure_file("${CMAKE_SOURCE_DIR}/examples/alanine-X.json" ${CMAKE_BINARY_DIR} COPYONLY)
-
-# Add profiling information for the appropriate examples
-if (ENABLE_PROFILING)
-  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-  target_compile_options("soap_profile" PRIVATE -ggdb)
-    # target_compile_options("soap_profile" PRIVATE -pg)
-    # set_target_properties("soap_profile" PROPERTIES LINK_FLAGS -pg)
+if (${BUILD_EXAMPLES})
+  # compile exmaples executables
+  file( GLOB binaries "${CMAKE_SOURCE_DIR}/examples/*.cc")
+  foreach(binaryfile ${binaries})
+    get_filename_component(binaryname  ${binaryfile} NAME_WE)
+    add_executable(${binaryname} ${binaryfile})
+    target_link_libraries(${binaryname} "${LIBRASCAL_NAME}")
+  endforeach(binaryfile ${binaries})
+  configure_file("${CMAKE_SOURCE_DIR}/examples/crystal_structure.json" ${CMAKE_BINARY_DIR} COPYONLY)
+  configure_file("${CMAKE_SOURCE_DIR}/examples/alanine-X.json" ${CMAKE_BINARY_DIR} COPYONLY)
+
+  # Add profiling information for the appropriate examples
+  if (ENABLE_PROFILING)
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    target_compile_options("soap_profile" PRIVATE -ggdb)
+      # target_compile_options("soap_profile" PRIVATE -pg)
+      # set_target_properties("soap_profile" PROPERTIES LINK_FLAGS -pg)
```

Comment:
are the uncommented still needed? what does -pg give you in addition to -ggdb? I only read in the doc that -pg disables the -Ox flags which you probably don't want here?

Reply:
The `-pg` option is unrelated to `-ggdb`; the former adds profiling information for `gprof`. I don't think we're using that anymore, though, IIRC. In any case, these commented lines can be removed.